SSA/Projects/InstCombine/AliveStatements.lean

```diff
@@ -13,6 +13,11 @@ open LLVM
     getLsb (-1#w) ↑i :=
   BitVec.getLsb_negOne ..
 
+@[simp]
+lemma BitVec.ofNat_toNat_zero : 
+BitVec.toNat (BitVec.ofInt w 0) = 0 := 
+  by simp[BitVec.toNat, BitVec.ofInt, BitVec.toFin, BitVec.ofNat, OfNat.ofNat]
```

Comment:
This was already added to SSA/Projects/InstCombine/AliveAutoGenerated.lean, no?

Reply:
Btw, is that a lemma that we can propose for std or mathlib?

Reply:
It might be, I have not checked if upstream has it, would be a good idea to check.

Reply:
But upstream would probably like a more general theorem than this one.

Reply:
https://github.com/leanprover/std4/blob/main/Std/Data/BitVec/Lemmas.lean#L59C1-L59C1

> theorem toNat_ofNat (x w : Nat) : (BitVec.ofNat w x).toNat = x % 2^w := by

So sth similar exists. It might be useful to think if there is sth to upstream here. Maybe @alexkeizer has some advise.

Reply:
First generalization could be `BitVec.toNat (BitVec.ofInt w <| BitVec.ofNat x) = x`, but I'm not sure if `simp` will recognize that `0 : Int` unifies with `BitVec.ofNat 0`. We could also go even further to
```lean
BitVec.toNat (BitVec.ofInt w i) 
= match i with
   | .ofNat x = x
   | .negSucc x = _ // something with modulo or inverses or whatever
```